keps/sig-node/3063-dynamic-resource-allocation/kep.yaml

```diff
@@ -6,7 +6,7 @@ owning-sig: sig-node
 participating-sigs:
   - sig-scheduling
   - sig-autoscaling
-status: implementable
+status: withdrawn
```

Comment:
You should probably define what "withdrawn" means.

Are we removing everything related to classic DRA in 1.32?

Are we deprecating the feature and removing in a future release (say 1.33 or 1.34?)?

Given that there has been some ask to keep this around, I wonder if we should deprecate this feature and announce that it is deprecated and will be removed in a future release may be the least bad option.

Reply:
`withdrawn` is a constant in the KEP process. I took it from the template, but I don't know where it is defined.

> Are we removing everything related to classic DRA in 1.32?

Yes.

> Given that there has been some ask to keep this around, I wonder if we should deprecate this feature and announce that it is deprecated and will be removed in a future release may be the least bad option.

The intent to remove it has been communicated several times. Because it's an alpha feature, we can remove it without a deprecation period.

Reply:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/0000-kep-process/README.md#kep-workflow

> withdrawn: The authors have withdrawn the KEP.

This is one of those circular definitions which don't add any information... "Withdrawn means that it's withdrawn".

Reply:
I can also use "rejected".

Reply:
My concern is that there has not really been any kind of formal communication that this will be dropped. 

It is alpha so I guess this is fine. But normally I would expect to see some kind of deprecation in the release notes and documentation stating that this will be going away in a future release. It would make me unhappy as a user to see something in k8s for a few releases and then in the latest it was dropped without any warning in documentation/release-notes.

I guess for alpha we can accept that. I just wonder if we should go with a deprecation process followed by removal as I don't know if we have any documentaiton announcing that this is going away other than github issues.

Reply:
> I can also use "rejected".

Is there a deprecated option?

Reply:
> My concern is that there has not really been any kind of formal communication that this will be dropped.

It has been mentioned in presentations, Slack and working group meetings. If someone has been using this feature and not paid any attention, then that's of course unfortunate, but I don't think we should go out of our way and keep it around longer because of that.

I'm not the one asking for the removal and it feels a bit odd that I now need to defend that decision, but alpha is alpha for a reason! :sweat: 

> Is there a deprecated option?

No.

Reply:
I have not seen us deprecate alpha features before so maybe @johnbelamaric, @thockin or @liggitt can weigh in on the approach for removing.

Reply:
As Patrick says, alpha is alpha for a reason and this was heavily discussed in multiple forum for a long time, ... we just remove the code https://github.com/kubernetes/kubernetes/pull/121229 all together

Reply:
Alright. Withdrawn seems to be the right state and we can remove without deprecating.

Reply:
Withdrawn is correct, and we have done this before (e.g. topology keys).

Reply:
Yes. And no deprecation is needed for alpha. I think this is the right decision. Experimentation can be done in 1.31 and earlier releases; any learning from that can be incorporated into new features built on our structured parameters base. There is no need to keep the experimentation available in 1.32 any longer.

Reply:
Off topic, but release team seems to also be a bit confused on our policy. Is this written down anywhere?